pandas/io/formats/style.py

```diff
@@ -1796,6 +1800,163 @@ def f(data: FrameOrSeries, props: str) -> np.ndarray:
             props = f"background-color: {color};"
         return self.apply(f, axis=axis, subset=subset, props=props)
 
+    def highlight_between(
+        self,
+        subset: Optional[IndexLabel] = None,
+        color: str = "yellow",
+        axis: Optional[Axis] = 0,
+        left: Optional[Union[Scalar, Sequence]] = None,
+        right: Optional[Union[Scalar, Sequence]] = None,
+        inclusive: Union[bool, str] = True,
+        props: Optional[str] = None,
+    ) -> Styler:
+        """
+        Highlight a defined range with a style.
+
+        .. versionadded:: 1.3.0
+
+        Parameters
+        ----------
+        subset : IndexSlice, default None
+            A valid slice for ``data`` to limit the style application to.
+        color : str, default 'yellow'
+            Background color to use for highlighting.
+        axis : {0 or 'index', 1 or 'columns', None}, default 0
+            If ``left`` or ``right`` given as sequence, axis along which to apply those
+            boundaries. See examples.
+        left : scalar or datetime-like, or sequence or array-like, default None
+            Left bound for defining the range.
+        right : scalar or datetime-like, or sequence or array-like, default None
+            Right bound for defining the range.
+        inclusive : {'both', 'neither', 'left', 'right'} or bool, default True
```

Comment:
can we not include bool and just have a sensisble default? both?

Reply:
of course, I included `bool` for consistency with `Series.between` which only accepts `bool`, and since I thought it was a natural expectation to the `inclusive?` argument. Its inclusion didn't seem to big factor in code.

changed.